### PR TITLE
Add bazel rule to build C# release artifacts

### PR DESCRIPTION
**Pull request by deannagarcia**
The new release script will build both Google.Protobuf and Google.Protobuf.Tools. Installing nuget/dotnet will be done in the docker image and setting up the necessary executable paths for Google.Protobuf.Tools will be done in internal preprocessing.

### CODE REVIEW

File: csharp/build_release.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+cd $(dirname $(readlink $BASH_SOURCE))
+
+#Disable some unwanted dotnet options
```
Comment (jskeet) [inline csharp/build_release.sh:5]: Tiny nit-pick: I would suggest having a space after the `#` in each comment line, for readability. (Having looked at a couple of shell scripts in the root directory, it looks like that's common in this repo too.)

**Review by jskeet — APPROVED** [review]
Just one tiny nit-pick that I've suggested for comments.

File: csharp/build_release.sh (hunk 1 above)
Comment (deannagarcia) [reply csharp/build_release.sh]: Fixed!

File: csharp/BUILD.bazel [hunk 2]
```diff
@@ -51,3 +52,9 @@ pkg_files(
     strip_prefix = strip_prefix.from_root(""),
     visibility = ["//pkg:__pkg__"],
 )
+
+sh_binary(
+  name = "release",
+  srcs = ["build_release.sh"],
+  args = ["$(location build_release.sh)"],
```
Comment (perezd) [inline csharp/BUILD.bazel:59]: what is this for? I don't see it used in the script

**Review by perezd — APPROVED** [review]
LGTM, Are there already changes in the base docker image for nuget/donet binaries?

**Comment by deannagarcia** [comment]
> LGTM, Are there already changes in the base docker image for nuget/donet binaries?

Yep, they were already added!

File: csharp/BUILD.bazel (hunk 2 above)
Comment (deannagarcia) [reply csharp/BUILD.bazel:59]: This is the bazel target to run the bash script so we can just `bazel run //csharp:release`

**Pull request merged**